hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/TraceEntityAccessorBuilder.java

```diff
@@ -35,13 +38,19 @@ public TraceEntityAccessorBuilder withExcludeEntityTypes(Set<String> excludedEnt
     return this;
   }
 
+  public TraceEntityAccessorBuilder withExecutor(Executor executor) {
+    this.executor = executor;
+    return this;
+  }
+
   public TraceEntityAccessor build() {
     return new DefaultTraceEntityAccessor(
         this.entityTypeClient,
         this.entityDataClient,
         this.attributeClient,
         TraceAttributeReaderFactory.build(this.attributeClient),
         entityWriteThrottleDuration,
-        excludedEntityTypes);
+        excludedEntityTypes,
+        Schedulers.from(executor));
```

Comment:
Do we have access to the shutdown hook here? for a manually created scheduler, we should call shutdown when the app is shutting down.